operator/pkg/ciliumendpointslice/cescache.go

```diff
@@ -74,6 +101,51 @@ func (c *CESCache) deleteNode(nodeName NodeName) []CESKey {
 	return nil
 }
 
+// Insert a CID into the cache and return the CESs which need to be reconciled
+func (c *CESCache) insertCID(cid CID, gidLabels Label) []CESKey {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	// Clean up old state
+	if oldLabels, ok := c.cidToGidLabels[cid]; ok && oldLabels != gidLabels {
+		c.globalIdLabelsToCIDSet[oldLabels].ids.Delete(cid)
+		// If the selectedID is the same as the CID being removed, update it to another valid CID, if it exists.
+		if c.globalIdLabelsToCIDSet[oldLabels].selectedID == cid {
+			c.globalIdLabelsToCIDSet[oldLabels].setSelectedID("", true)
+		}
```

Comment:
I assume one should also return the `CESKey` entries touched here, as they'd also need to be reconciled again.